hnn_core/tests/conftest.py

```diff
@@ -0,0 +1,59 @@
+""" Example from pytest documentation
+
+https://pytest.org/en/stable/example/simple.html#incremental-testing-test-steps
+"""
+
+from typing import Dict, Tuple
+import pytest
+
+# store history of failures per test class name and per index in parametrize
+# (if parametrize used)
+_test_failed_incremental: Dict[str, Dict[Tuple[int, ...], str]] = {}
+
+
+def pytest_runtest_makereport(item, call):
+
+    if "incremental" in item.keywords:
+        # incremental marker is used
+        if call.excinfo is not None and not call.excinfo.typename == "Skipped":
```

Comment:
I added `and not call.excinfo.typename == "Skipped"` so that skipped tests (i.e. `mpi4py` not being installed) would not cause subsequent tests in the incremental block to be xfailed.

Reply:
> humm ... how should we interpret xfailed? Will it cause CI to fail?
> 
It's the same as failed. CI should fail if MPI tests fail, no?

> also, can you add a comment somewhere indicating that this is the only line you modified and everything else is copied. Will make it easier to maintain the code.

No problem

Reply:
Note that xfailed in this case will only follow a true failure. CI will fail.